mlabs/test/Test/NFT/QuickCheck.hs

```diff
@@ -0,0 +1,65 @@
+module Test.NFT.QuickCheck where
+
+import PlutusTx.Prelude hiding (fmap, length, mconcat, (<$>), (<*>))
+import Prelude (
+  div,
+  fmap,
+  mconcat,
+  (<$>),
+  (<*>),
+ )
+
+import Control.Monad.Reader (lift)
+import Data.Bool (bool)
+import Plutus.Contract.Test (TracePredicate, Wallet (..), checkPredicateInner)
+import Plutus.Trace.Emulator (EmulatorRuntimeError (..), throwError)
+import Test.QuickCheck qualified as QC
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.QuickCheck (testProperty)
+
+import Mlabs.Emulator.Scene (Scene, checkScene)
+import Mlabs.NFT.Validation
+import Test.NFT.Init
+
+positiveInteger :: QC.Gen Integer
+positiveInteger = fmap ((* 100) . QC.getPositive) (QC.resize 5 QC.arbitrary)
+
+wallets :: [Wallet]
+wallets = [w1, w2, w3]
+
+mkScript :: ([Wallet], Integer) -> Script
+mkScript ([w1', w2', w3'], price) = do
+  userAct w1' $ SetPriceAct (Just price) "NFT"
+  userAct w2' $ BuyAct price Nothing "NFT"
+  userAct w2' $ SetPriceAct (Just (price * 2)) "NFT"
+  userAct w3' $ BuyAct (price * 2) Nothing "NFT"
+mkScript _ = lift $ throwError $ GenericError "Unreachable"
+
+mkScene :: ([Wallet], Integer) -> Scene
+mkScene ([w1', w2', w3'], price) =
+  mconcat
+    [ w1' `ownsAda` (price1 + share1 + share2)
+    , w2' `ownsAda` (- price1 - share1 + price2)
+    , w3' `ownsAda` (- price2 - share2)
+    ]
+  where
+    price1 = price - share1
+    share1 = price `div` 10
+    price2 = price * 2 - share2
+    share2 = (price * 2) `div` 10
+mkScene _ = noChangesScene
+
+testContract :: QC.Property
+testContract = QC.forAll ((,) <$> QC.shuffle wallets <*> positiveInteger) testProp
```

Comment:
I think that we want "proper" QC testing as per plutus pioneer -- take a look at it @t4ccer and try to make it work, we want this project to be the standard setter

Reply:
Ok, will take care of it